### ISSUE TITLE
[a11y]: Icon in Small Toggle Causing Violation (Doesn't use aria-hidden or aria-label)

### ISSUE THREAD

**Issue by RachelGHogan** [carbon-design-system/carbon#17598]
### Package

@carbon/react

### Operating System

MacOS

### Package version

1.66.0

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker v3.1.75.9999

### Description

When a small toggle is clicked, a checkmark icon appears on the toggle which is setting off the accessibility scanner, as it doesn't have a label or aria-hidden set to true like the rest of the carbon icons. Violation shown below in storybook and stackblitz: 
![image](https://github.com/user-attachments/assets/f26ba2c4-ea6d-49dc-a5e6-262a681bee59)
![image](https://github.com/user-attachments/assets/677e59c5-90e6-4983-890a-da8bf7a059a9)

### Reproduction/example

https://stackblitz.com/edit/github-ba7pgw-dsvrxa?file=src%2FApp.jsx

### Steps to reproduce

1. Open the stackblitz link from the above reproduction section
2. Open preview in a new tab (the accessibility scanner doesn't work well otherwise)
3. Toggle the button to the on position so the checkmark becomes visible
4. Run the accessibility scanner

### Suggested Severity

None

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by alina-jacob** [comment]
Duplicate issue: https://github.com/carbon-design-system/carbon/issues/17223

**Comment by 2nikhiltom** [comment]
Hey! It seems you'll need to open the component in a separate iframe and run an accessibility scan. This ensures only the component is tested. I didn't encounter the issue during my scan. The violations you're seeing might be caused by testing it directly in Storybook.

https://github.com/user-attachments/assets/7d5e818c-c9f8-469a-8b9c-d06eee1782c4
